### PR TITLE
Remove QuoteStr() usage

### PR DESCRIPTION
**Pull request by BetaCat0**
As above

### CODE REVIEW

File: engine.go [hunk 1]
```diff
@@ -196,13 +197,28 @@ func (engine *Engine) Quote(value string) string {
...
 	}
 
-	if string(value[0]) == engine.dialect.QuoteStr() || value[0] == '`' {
+	quotePair := engine.dialect.Quote("")
+
+	if value[0] == '`' || len(quotePair) != 2 || value[0] == quotePair[0] { // no quote
 		return value
-	}
+	} else {
+		if len(quotePair) != 2 {
```
Comment (lunny) [inline engine.go]: duplicated code with line 202

File: engine.go [hunk 2]
```diff
@@ -216,20 +232,35 @@ func (engine *Engine) QuoteTo(buf *builder.StringBuilder, value string) {
...
-	if string(value[0]) == engine.dialect.QuoteStr() || value[0] == '`' {
-		buf.WriteString(value)
+	quotePair := engine.dialect.Quote("")
+
+	if value[0] == '`' || len(quotePair) != 2 || value[0] == quotePair[0] { // no quote
+		_, _ = buf.WriteString(value)
 		return
-	}
+	} else {
+		if len(quotePair) != 2 {
```
Comment (lunny) [inline engine.go]: As above.

File: engine.go [hunk 3]
```diff
@@ -216,20 +232,35 @@ func (engine *Engine) QuoteTo(buf *builder.StringBuilder, value string) {
...
+			return
+		}
 
-	value = strings.Replace(value, ".", engine.dialect.QuoteStr()+"."+engine.dialect.QuoteStr(), -1)
+		prefix, suffix := quotePair[0], quotePair[1]
 
-	buf.WriteString(engine.dialect.QuoteStr())
-	buf.WriteString(value)
-	buf.WriteString(engine.dialect.QuoteStr())
+		_ = buf.WriteByte(prefix)
```
Comment (lunny) [inline engine.go]: The code is duplicated with them on `Quote`, could you create a new function for these?

Comment (BetaCat0) [reply engine.go]: Oh, I see. Please wait

File: session_insert.go [hunk 4]
```diff
@@ -242,23 +242,17 @@ func (session *Session) innerInsertMulti(rowsSlicePtr interface{}) (int64, error
...
 	var sql string
 	if session.engine.dialect.DBType() == core.ORACLE {
-		temp := fmt.Sprintf(") INTO %s (%v%v%v) VALUES (",
+		temp := fmt.Sprintf(") INTO %s (%v) VALUES (",
 			session.engine.Quote(tableName),
-			session.engine.QuoteStr(),
-			strings.Join(colNames, session.engine.QuoteStr()+", "+session.engine.QuoteStr()),
-			session.engine.QuoteStr())
-		sql = fmt.Sprintf("INSERT ALL INTO %s (%v%v%v) VALUES (%v) SELECT 1 FROM DUAL",
+			session.engine.Quote(strings.Join(colNames, session.engine.Quote(", "))))
```
Comment (lunny) [inline session_insert.go]: Before code assumed the two quotes is equal, but this one is not. So `session.engine.Quote(", ")` is not correct.

Comment (BetaCat0) [reply session_insert.go]: Got it

File: statement.go [hunk 5]
```diff
@@ -581,7 +581,8 @@ func (statement *Statement) col2NewColsWithQuote(columns ...string) []string {
 	newColumns := make([]string, 0)
 	for _, col := range columns {
 		col = strings.Replace(col, "`", "", -1)
```
Comment (lunny) [inline statement.go]: This line also could be removed.

Comment (BetaCat0) [reply statement.go]: Okay

**Comment by lunny** [comment]
Thanks!

**Pull request merged**